libcxx/test/benchmarks/exception_ptr.bench.cpp

```diff
@@ -18,4 +18,118 @@ void bm_make_exception_ptr(benchmark::State& state) {
 }
 BENCHMARK(bm_make_exception_ptr)->ThreadRange(1, 8);
 
+void bm_exception_ptr_copy_ctor_nonnull(benchmark::State& state) {
+  std::exception_ptr excptr = std::make_exception_ptr(42);
+  for (auto _ : state) {
+    benchmark::DoNotOptimize(std::exception_ptr(excptr));
+  }
+}
+BENCHMARK(bm_exception_ptr_copy_ctor_nonnull);
+
+void bm_exception_ptr_copy_ctor_null(benchmark::State& state) {
+  std::exception_ptr excptr = nullptr;
+  for (auto _ : state) {
+    std::exception_ptr excptr_copy(excptr);
+    benchmark::DoNotOptimize(excptr_copy);
+    // The compiler should be able to constant-fold the comparison
```

Comment:
Since the `excptr_copy` escapes through `DoNotOptimize`, this comment is incorrect. Throughout.